### PR TITLE
[FINERACT-1958] Business events for down payment installments

### PR DESCRIPTION
**Pull request by taskain7**
## Description

Installment data in events include if the isntallment is down payment.
LoanTransactionEvenet is raised on auto payment on down paymnet.
API changes if the installment is down payment.

### CODE REVIEW

**Review by ruchiD — COMMENTED** [review]
LGTM

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanAccountDomainServiceJpa.java [hunk 1]
```diff
@@ -313,7 +311,7 @@ private LoanTransactionBusinessEvent getTransactionRepaymentTypeBusinessEvent(Lo
         } else if (isRecoveryRepayment) {
             repaymentEvent = new LoanTransactionRecoveryPaymentPostBusinessEvent(transaction);
         } else if (repaymentTransactionType.isDownPayment()) {
-            repaymentEvent = new LoanTransactionDownPaymentPostBusinessEvent(transaction);
+            repaymentEvent = new LoanTransactionMakeRepaymentPostBusinessEvent(transaction);
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanAccountDomainServiceJpa.java]: Are you sure? we dont wanna trigger downpayment business event for downpayment transaction?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanAccountDomainServiceJpa.java [hunk 2]
```diff
@@ -292,7 +290,7 @@ private LoanBusinessEvent getLoanRepaymentTypeBusinessEvent(LoanTransactionType
         } else if (isRecoveryRepayment) {
             repaymentEvent = new LoanTransactionRecoveryPaymentPreBusinessEvent(loan);
         } else if (repaymentTransactionType.isDownPayment()) {
-            repaymentEvent = new LoanTransactionDownPaymentPreBusinessEvent(loan);
+            repaymentEvent = new LoanTransactionMakeRepaymentPreBusinessEvent(loan);
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanAccountDomainServiceJpa.java]: Are you sure? we dont wanna trigger downpayment business event for downpayment transaction?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java [hunk 3]
```diff
@@ -456,7 +458,14 @@ public CommandProcessingResult disburseLoan(final Long loanId, final JsonCommand
                 changedTransactionDetail = loan.disburse(currentUser, command, changes, scheduleGeneratorDTO, null, downPaymentEnabled);
             }
             loan.adjustNetDisbursalAmount(amountToDisburse.getAmount());
-            loan.handleDownPayment(amountToDisburse.getAmount(), command, scheduleGeneratorDTO);
+            if (loan.isAutoRepaymentForDownPaymentEnabled()) {
+                businessEventNotifierService.notifyPreBusinessEvent(new LoanTransactionMakeRepaymentPreBusinessEvent(loan));
+                LoanTransaction downPaymentTransaction = loan.handleDownPayment(amountToDisburse.getAmount(), command,
+                        scheduleGeneratorDTO);
+                businessEventNotifierService
+                        .notifyPostBusinessEvent(new LoanTransactionMakeRepaymentPostBusinessEvent(downPaymentTransaction));
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java]: Are you sure? we dont wanna trigger downpayment business event for downpayment transaction?

**Review by galovics — APPROVED** [review]

**Pull request merged**